api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_online.py

```diff
@@ -18,17 +18,28 @@ class MeanVarianceAccumulator:
         Knuth, Art of Computer Programming, volume II
     """
 
-    def __init__(self, n_batches: int, n_samples: npt.NDArray[np.int64], n_variables: int, ddof: int = 1):
+    def __init__(
+        self,
+        n_batches: int,
+        n_samples: npt.NDArray[np.int64],
+        n_variables: int,
+        ddof: int = 1,
+        exclude_zeros: bool = False,
+    ):
         if n_samples.sum() <= 0:
             raise ValueError("No samples provided - can't calculate mean or variance.")
 
+        self.exclude_zeros = exclude_zeros
         self.ddof = ddof
         self.n_batches = n_batches
         self.n_samples = n_samples
         self.n = np.zeros((n_batches, n_variables), dtype=np.int32)
         self.u = np.zeros((n_batches, n_variables), dtype=np.float64)
         self.M2 = np.zeros((n_batches, n_variables), dtype=np.float64)
 
+        if self.exclude_zeros and self.n_batches > 1:
```

Comment:
Implementing `exclude_zeros` with batching requires some nontrivial changes to `_mbomv_combine_batches`, so since we're not using it, I decided to disable it. We can enable it later if necessary.

Reply:
Batching seems pretty fundamental to the use case - @pablo-gar what are your thoughts? 
 Without it, any use case that requires batching will have to do batching above this API, requiring multiple passes through the data.

I"m also unclear what complexity is implied by adding multi-batch support to the combine batches step.  I believe that the only change is to not adjust the value of `n` for the total sample count (in `n_samples`), which is a 1-2 line change....

Reply:
Will implement as part of https://github.com/chanzuckerberg/cellxgene-census/issues/786

Reply:
Some extra context. Batching has a bit of a different connotation here as compared to highly variable genes, in highly variable genes average/variance are calculated per batch and the results are aggregated to retrieve _batch-corrected_ variable genes. In this case the functionality that we want to enable (which I believe would be enabled by this option) is a "group_by" functionality to save user the hassle to call this function in a for loop.

Specifically, as an example a user may want perform an axis query to obtain all cells from a tissue and _then_ perform average gene expression for each cell types a.k.a "group by".

Agreed to add this in a different PR

Reply:
I don't see a use case for "batch-corrected" mean and variance calculations.

Reply:
We are using the 'batch' term in too many ways!  The thing we are advocating for has nothing to do with batch correction.

What the current underlying implementation provides is a way to perform separate mean/var calculations, on subsets of the data identified by a batch key, all in one pass.  This is a performance win, and very important for algorithms that need the stats per-batch in order to do other work (e.g.,. HVG uses this)

If we don't add this, then a user that needs per-batch stats will have to do multiple passes over the data (perf hit)